src/Service/QuizInstanceService.php

```diff
@@ -128,13 +129,28 @@ public function getResultsNumber(Criteria $criteria): int
         return $quizInstances->getCount();
     }
 
+    /**
+     * Marks the completion of a quiz in the database.
+     *
+     * @param int $quizInstanceId
+     * @param bool $isComplete
+     */
+    public function markQuizComplete(int $quizInstanceId, bool $isComplete): void
+    {
+        /** @var $quizInstance QuizInstance */
+        $quizInstance = $this->quizInstanceRepo->find($quizInstanceId);
+        $quizInstance->setIsCompleted($isComplete);
+
+        $this->quizInstanceRepo->insertOnDuplicateKeyUpdate($quizInstance);
```

Comment:
isn't this the same as `quizInstance->save()` (if yes, I suggest this approach, it is cleaner)